### PR TITLE
WIP: Python basics

### PR DESCRIPTION
**Pull request by alexpearce**
This is a start on the Python basics lessons, going over shells, objects, and the builtin types.

We'll probably want to teach faster than the pace of the written lessons if we want to include third-party libraries like numpy.

What's left:

- [x] Conditionals
- [x] Writing methods
- [x] `import`
- [x] Scripting (`{,i}python` and `{,i}python -i`)
- [ ] Writing classes? This went pretty badly when I tried to teach it on-the-fly
- [x] Modules?
- [x] Finding help, other cool builtins
- [x] Conventional coding

@gdujany is teaching one group; are the other two instructors on GitHub? Would be good to start getting feedback.

### CODE REVIEW

**Comment by chrisburr** [comment]
I think @mconcas is? And I can't find Gedas on GitHub but I'll point him to this PR.

**Comment by mconcas** [comment]
Yes, here I am. 
Currently I am at the Bertinoro computing school. As soon as it finishes (Saturday morning) I will have a look.

**Comment by alexpearce** [comment]
I don't think I'll have time to write anything about third-party packages. How much of a problem do we think this might be?

File: python/running.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+$ which python
+/cvmfs/lhcb.cern.ch/lib/lcg/releases/LCG_88/Python/2.7.13/x86_64-slc6-gcc49-opt/bin/python
+$ python --version
+Python 2.7.13
+```
+
+We can add a few Python utilities to our environment by sourcing a setup file:
+
+```bash
+$ source /cvmfs/sft.cern.ch/lcg/views/LCG_88/$CMTCONFIG/setup.sh
```
Comment (chrisburr) [inline python/running.md]: This doesn't work with `zsh` due to bugs in the LCG setup script (`PYTHONPATH` is set incorrectly but there could be other issues). It's probably best that we make sure the helpers are aware just get everyone to use bash.

Comment (alexpearce) [reply python/running.md]: Ah, bummer. Yes, we should then.

**Comment by alexpearce** [comment]
I've completed most of the lessons on basic Python. I haven't done writing your own modules yet. What do we think about that topic?

**Comment by alexpearce** [comment]
Preview here: https://apearce.web.cern.ch/apearce/analysis-essentials-f1e44ac/python/introduction.html

**Comment by dberzano** [comment]
@alexpearce I think writing your own modules is an "advanced" topic, but on the other hand it's very simple to explain and teaches developers to organize their code properly, and in a way that others can reuse it. I would add it 👍

**Comment by dberzano** [comment]
By the way do you want us to take care of it maybe?

**Comment by alexpearce** [comment]
Adding info on how to create your own modules? That would be great! Would also be cool to somehow integrate it with the scripts lesson, so that we can teach the meaning of `if __name__ == '__main__'`.

**Comment by chrisburr** [comment]
As the lesson is tomorrow I am going to merge this now, any further changes can be put into a new PR.

**Comment by chrisburr** [comment]
As the lesson is tomorrow I am going to merge this now, any further changes can be put into a new PR.

Thank you to everyone involved in putting this together 🎉

**Comment by dberzano** [comment]
Yup a new PR is coming soon (again, sorry for the bad timing). Changes are not radical though.

**Comment by dberzano** [comment]
fyi #25.

**Pull request merged**